tests/GamepadService-spec.js

```diff
@@ -112,3 +114,84 @@ describe("Consumer", () => {
     expect(consumer.hasGesture).toBe(false);
   });
 });
+
+describe("GamingDevice", () => {
+  const gamingDevices = document.createElement("div");
+  document.body.append(gamingDevices);
+
+  beforeEach(() => {
+    gamingDevices.innerHTML = "";
+  });
+
+  afterAll(() => {
+    gamingDevices.remove();
+  });
+
+  it("can be constructed with default attributes ", () => {
+    const gamingDevice = new GamingDevice();
+
+    expect(gamingDevice).toBeTruthy();
+
+    expect(gamingDevice).toBeTruthy();
+    expect(Number.isInteger(gamingDevice.id)).toBe(true);
+    expect(gamingDevice.connected).toBe(false);
+    expect(gamingDevice.buttons.length).toEqual(18);
+  });
+
+  it("has the correct number of buttons when number of buttons is specified", () => {
+    const gamingDevice = new GamingDevice(10);
+
+    expect(gamingDevice).toBeTruthy();
+
+    expect(gamingDevice).toBeTruthy();
+    expect(Number.isInteger(gamingDevice.id)).toBe(true);
+    expect(gamingDevice.connected).toBe(false);
+    expect(gamingDevice.buttons.length).toEqual(10);
+  });
+
+  it("can be constructed with default attributes and contains the correct html elements", () => {
+    const gamepadDevice = GamingDevice.attachGamingDevice(gamingDevices);
+
+    expect(gamepadDevice.parentElement).toBe(gamingDevices);
+
+    expect(gamepadDevice).toBeTruthy();
+    expect(Number.isInteger(gamepadDevice.id)).toBe(true);
+    expect(gamepadDevice.connected).toBe(false);
+    expect(gamepadDevice.buttons.length).toEqual(18);
+    expect(gamepadDevice.children[1].childElementCount).toEqual(18);
+  });
+
+  it("changes its connected status to true when any one of its buttons is clicked", () => {
+    const gamingDevice = new GamingDevice();
+
+    expect(gamingDevice.connected).toEqual(false);
+
+    const buttonIndex = Math.floor(Math.random() * gamingDevice.buttons.length);
+    gamingDevice.buttons[buttonIndex].click();
+
+    expect(gamingDevice.connected).toEqual(true);
+  });
+
+  it("is an extended HTMLDivElement", () => {
+    const gamingDevice = new GamingDevice();
+
+    expect(gamingDevice.tagName).toEqual("DIV");
+  });
+});
+
+describe("GamingDeviceButton", () => {
+  it("can be constructed with default GamingDeviceButton attributes", () => {
+    const button = new GamingDeviceButton();
+
+    expect(button).toBeTruthy();
+    expect(button.pressed).toBe(false);
+    expect(button.touched).toBe(false);
+    expect(button.value).toBe(0.0);
+  });
+
+  it("is an extended HTMLButtonElement", () => {
+    const button = new GamingDeviceButton();
+
+    expect(button.tagName).toEqual("BUTTON");
```

Comment:
```suggestion
    expect(button.tagName).toBe("BUTTON");
```
Remember .toEqual() is "match", but we want `.toBe()`... I know, very Shakespeare 😂 

However, I think we want: 
```JS
   expect(button instanceof HTMLButtonElement).toBe(true)
```

To check if it "extends" properly.